### ISSUE TITLE
Aborted (core dumped) in `flow.nn.BCELoss` when logits contain negative values

### ISSUE THREAD

**Issue by deepliao** [Oneflow-Inc/oneflow#10639]
## Summary

Using `flow.nn.BCELoss` on unbounded input logits (e.g., raw outputs from `flow.randn()`) leads to a **segmentation fault (core dumped)**. The BCELoss operator internally expects the input tensor to contain probabilities in the range **[0, 1]**. However, OneFlow does not enforce or validate this requirement at the Python level. Passing unbounded inputs (e.g., values < 0) causes a crash in the C++ kernel instead of raising a clear Python exception like `ValueError`.

## Code to reproduce bug

```python
import oneflow as flow
import numpy as np

# Set seed
np.random.seed(42)

def generate_test_data(batch_size=3, numFeatures=4, seed=42):
    logits = flow.tensor(np.random.randn(batch_size, numFeatures), dtype=flow.float32)
    target = flow.tensor(np.random.rand(batch_size, numFeatures), dtype=flow.float32)
    return logits, target

# Generate test data
logits, target = generate_test_data()

# (Optional but incorrect) Applying sigmoid to target instead of logits
target = flow.sigmoid(target)

# This line causes a crash due to invalid logits (raw values not in [0, 1])
loss = flow.nn.BCELoss(reduction='none')(logits, target)

print("Loss:", loss)
```

output:
```
*** Check failure stack trace: ***
    @     0x7f27b9b959ca  google::LogMessage::Fail()
    @     0x7f27b9b95cb2  google::LogMessage::SendToLog()
    @     0x7f27b9b95537  google::LogMessage::Flush()
    @     0x7f27b9b980a9  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f27b3db54a9  oneflow::user_op::(anonymous namespace)::BinaryCrossEntropyKernel<>::Compute()
    @     0x7f27b591354d  oneflow::one::StatefulOpKernel::Compute()
    @     0x7f27b3badcab  oneflow::vm::OpCallInstructionUtil::Compute()
    @     0x7f27b3bab787  oneflow::vm::OpCallInstructionPolicy::Compute()
    @     0x7f27b3ba75bc  oneflow::vm::Instruction::Compute()
    @     0x7f27b3ba5a6f  oneflow::vm::EpStreamPolicyBase::Run()
    @     0x7f27b3bb1086  oneflow::vm::StreamPolicy::RunIf()
    @     0x7f27b3bb86de  oneflow::vm::ThreadCtx::TryReceiveAndRun()
    @     0x7f27b3bbad2d  oneflow::(anonymous namespace)::WorkerLoop()
    @     0x7f27b3bbb11f  _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJPFvPN7oneflow2vm9ThreadCtxERKSt8functionIFvS6_EEES6_ZNS3_14VirtualMachine15CreateThreadCtxENS3_6SymbolINS3_6DeviceEEENS3_10StreamTypeEmEUlS6_E3_EEEEE6_M_runEv
    @     0x7f27b9baa40f  execute_native_thread_routine
    @     0x7f2892a12609  start_thread
    @     0x7f2892939293  clone
Aborted (core dumped)
```

##  System Information
OneFlow installation: pip
OS: Ubuntu 20.04
OneFlow version (run python3 -m oneflow --doctor):
```
path: ['/root/miniconda3/envs/myconda/lib/python3.8/site-packages/oneflow']
version: 0.9.0
git_commit: 381b12c
cmake_build_type: Release
rdma: True
mlir: True
```
Python version: 3.8
CUDA driver version: 11.6
GPU models: NVIDIA A16
